server/apis/v1/response.go

```diff
@@ -16,17 +16,33 @@ limitations under the License.
 
 package v1
 
+import (
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+)
+
+var (
+	jsonMarshaller = new(runtime.JSONPb)
+)
+
 type NumaflowAPIResponse struct {
 	// ErrMsg provides more detailed error information. If API call succeeds, the ErrMsg is nil.
 	ErrMsg *string `json:"errMsg,omitempty"`
 	// Data is the response body.
-	Data interface{} `json:"data"`
+	Data any `json:"data"`
 }
 
 // NewNumaflowAPIResponse creates a new NumaflowAPIResponse.
-func NewNumaflowAPIResponse(errMsg *string, data interface{}) NumaflowAPIResponse {
+func NewNumaflowAPIResponse(errMsg *string, data any) NumaflowAPIResponse {
 	return NumaflowAPIResponse{
 		ErrMsg: errMsg,
 		Data:   data,
 	}
 }
+
+func (r NumaflowAPIResponse) MarshalJSON() ([]byte, error) {
+	m := map[string]any{
+		"errMsg": r.ErrMsg,
+		"data":   r.Data,
+	}
+	return jsonMarshaller.Marshal(m)
```

Comment:
Without using this particular JSON marshaller, the marshaled data will be something like `"watermarks":[{"value": "1719990590574"}]` instead of `"watermarks":["1719990590574"]`, that there's always a nested structure (`"value"`) for `google.protobuf.xxxx` data types.